components/service/glean-gradle-plugin/src/main/groovy/mozilla/components/service/glean-gradle-plugin/GleanGradlePlugin.groovy

```diff
@@ -0,0 +1,287 @@
+                    args "--allow-reserved"
+                }
+
+                doFirst {
+                    // Add the potential 'metrics.yaml' files at evaluation-time, rather than
+                    // configuration-time. Otherwise the Gradle build will fail.
+                    if (project.ext.has("allowMetricsFromAAR")) {
+                        inputs.files.forEach { file ->
+                            project.logger.lifecycle("Glean SDK - generating API from ${file.path}")
+                            args file.path
+                        }
+                    }
+                }
+
+                // Only show the output if something went wrong.
+                ignoreExitValue = true
+                standardOutput = new ByteArrayOutputStream()
+                errorOutput = standardOutput
+                doLast {
+                    if (execResult.exitValue != 0) {
+                        throw new GradleException("Process '${commandLine}' finished with non-zero exit value ${execResult.exitValue}:\n\n${standardOutput.toString()}")
+                    }
+                }
+            }
+
+            // Only attach the generation task if the metrics file is available or we're requested
+            // to fetch them from AAR files. We don't need to fail hard otherwise, as some 3rd party
+            // project might just want metrics included in Glean and nothing more.
+            if (project.file("${project.projectDir}/metrics.yaml").exists()
+                || project.file("${project.projectDir}/pings.yaml").exists()
+                || project.ext.has("allowMetricsFromAAR")) {
+                // This is an Android-Gradle plugin 3+-ism.  Culted from reading the source,
+                // searching for "registerJavaGeneratingTask", and finding
+                // https://github.com/GoogleCloudPlatform/endpoints-framework-gradle-plugin/commit/2f2b91476fb1c6647791e2c6fe531a47615a1e85.
+                // The added directory doesn't appear in the paths listed by the
+                // `sourceSets` task, for reasons unknown.
+                variant.registerJavaGeneratingTask(generateKotlinAPI, new File(sourceOutputDir))
+            }
+        }
+    }
+
+    void apply(Project project) {
+        // This installs a Miniconda3 environment into the gradle user home directory
+        // so that it will be shared between all libraries that use Glean.  This is
+        // important because it is approximately 300MB in installed size.
+        File condaBootstrapDir = new File(
+            project.getGradle().gradleUserHomeDir,
+            "glean/bootstrap-${MINICONDA_VERSION}"
+        )
+        File condaDir = new File(
+            condaBootstrapDir,
+            "Miniconda3"
+        )
+
+        // Define the path to the Miniconda3 installation as a buildConfig variable so
+        // that tests that need to run JSON through our JSON schema can do so.
+        project.android {
```

Comment:
This needs to ensure that your plugin depends on the Android plugin, which in turn can be tricky to arrange 'cuz the consuming project's Android plugin version can be different than a-c's.  I never found a great solution to this: see https://github.com/mozilla/application-services/pull/921#pullrequestreview-223342221 and the linked project.  This will definitely require more attention before landing, from me or from testers.

Reply:
You may find that this Just Works with Groovy and not with Kotlin.  Groovy is dynamic/at run-time; Kotlin is not.  Try to convert to Kotlin and you'll see the issues; I suggest you address them up front to avoid painting yourself into a corner.  (I painted myself into a corner.)

Reply:
It looks like this buildConfig variable is only needed by Glean itself, not by products using Glean, so it doesn't have to go in the plugin.  That neatly sidesteps this problem.